scripts/apps/search/directives/SearchResults.ts

```diff
@@ -424,8 +424,21 @@ export function SearchResults(
 
                 if (item) {
                     if (item._type === 'externalsource') {
-                        processPreview(item);
-                        return;
+                        var item_id = item.extra ? item.extra.itemid : false;
+
+                        if (item_id && item.guid) {
+                            criteria.params = {
+                                'detailed_info': {
+                                    'newsItemId': item.extra.itemid,
+                                    'guid': item.guid,
+                                }};
+                            return api.query(getProvider(criteria), criteria).then((item_) => {
+                                processPreview(item_._items[0]);
```

Comment:
I think we should check there is actually an item returned (and probably fallback to `processPreview(item);` call, the api might not work

Reply:
Ok, so can we do something like that?
```
return api.query(getProvider(criteria), criteria).then((item_) => {
           if (item_){
                item = item_._items[0] 
           }
           processPreview(item);    
});
```

Reply:
`_items[0]` could still be undefined, and there could be an error from the external server